erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -525,7 +525,10 @@ def set_missing_ref_details(
 						continue
 
 					if field == "exchange_rate" or not d.get(field) or force:
-						d.db_set(field, value)
+						if self.get("_action") in ("submit", "cancel"):
+							d.db_set(field, value)
+						else:
+							d.set(field, value)
```

Comment:
This is a bridge pattern and will disappear after a few years of using the document read only mode.